scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -102,30 +99,31 @@ class FormatWriter(formatOps: FormatOps) {
     Pattern.compile(raw"(?<=\n) *(?=[$pipe])", Pattern.MULTILINE)
   private val leadingPipeSpace = getStripMarginPattern('|')
 
-  private def formatMarginizedString(token: Token, indent: => Int): String = {
-    val pipeOpt =
-      if (!initStyle.assumeStandardLibraryStripMargin) None
-      else if (token.is[T.Interpolation.Part]) Some('|')
-      else getStripMarginChar(token)
-    pipeOpt.fold(token.syntax) { pipe =>
-      val pattern =
-        if (pipe == '|') leadingPipeSpace else getStripMarginPattern(pipe)
-      def firstChar: Char = token match {
-        case T.Interpolation.Part(_) =>
-          (for {
-            parent <- owners(token).parent
-            firstInterpolationPart <- parent.tokens.find(
-              _.is[T.Interpolation.Part]
-            )
-            char <- firstInterpolationPart.syntax.headOption
-          } yield char).getOrElse(' ')
-        case _ =>
-          token.syntax.find(_ != '"').getOrElse(' ')
-      }
-      val extraIndent =
-        if (initStyle.align.stripMargin && firstChar == pipe) 1 else 0
-      val spaces = getIndentation(indent + extraIndent)
-      pattern.matcher(token.syntax).replaceAll(spaces)
+  private def formatMarginized(ft: FormatToken, indent: => Int): String = {
+    val text = ft.left.syntax
+    val tupleOpt = ft.left match {
+      case _ if !initStyle.assumeStandardLibraryStripMargin => None
+      case _: T.Constant.String =>
+        getStripMarginChar(ft).map { pipe =>
+          (pipe, () => text.find(_ != '"').contains(pipe))
+        }
+      case _: T.Interpolation.Part =>
+        def check(x: FormatToken) = x.left.is[T.Interpolation.Start]
+        findToken(ft, prev)(check).right.toOption flatMap { t =>
+          getStripMarginChar(t).map { pipe =>
+            (pipe, () => t.right.syntax.headOption.contains(pipe))
+          }
+        }
+      case _ => None
+    }
+    tupleOpt.fold(text) {
+      case (pipe, isFirstCharPipe) =>
+        val pattern =
+          if (pipe == '|') leadingPipeSpace else getStripMarginPattern(pipe)
```

Comment:
Just propose to hide `if` inside the function

```scala
private def getStripMarginPattern(pipe: Char) =
  if (pipe == '|') leadingPipeSpace else getStripMarginPattern(pipe)

private val leadingPipeSpace = compileStripMarginPattern('|')

private def compileStripMarginPattern(pipe: Char) = 
  Pattern.compile(raw"(?<=\n) *(?=[$pipe])", Pattern.MULTILINE)
```

Reply:
will do, thank you